src/conf_mode/interfaces-tinc.py

```diff
@@ -0,0 +1,243 @@
+        tinc.update({'ed25516_private_key': ed25516_private_key, 'rsa_private_key': rsa_private_key});
+    
+    if 'deleted' in tinc:
+        tmp = leaf_node_changed(conf, ['network-dir'])
+        if tmp and len(tmp) == 1:
+            tinc.update({'network_dir':tmp[0]})
+    
+    return tinc
+
+def verify(tinc):
+    #bail out early - looks like removal from running config
+    if tinc is None:
+        return None
+    network = tinc['ifname']
+    if 'deleted' in tinc or 'disable' in tinc:
+        return None
+    if 'address' not in tinc:
+        raise ConfigError('address must be set')
+    if 'network_dir' not in tinc:
+        tinc.update({'network_dir': f'/config/tinc/{network}'})
+    if 'subnets' not in tinc:
+        raise ConfigError('subnets must be set')
+    if 'node_name' not in tinc:
+        raise ConfigError('node_name must be set')
+    if 'connect' in tinc:
+        node_name = tinc['node_name']
+        connect_peer_node_name = tinc['connect']
+        if node_name == connect_peer_node_name:
+            raise ConfigError('The local node name("{local_node}") and the remote target connection node name("{remote_node}") cannot match'.format(local_node = node_name, remote_node = connect_peer_node_name))
+    
+    if 'proxy' in tinc:
+        if 'address' not in tinc['proxy']:
+            raise ConfigError('proxy.address must be set')
+        if 'port' not in tinc['proxy']:
+            raise ConfigError('proxy.port must be set')
+        if 'type' not in tinc['proxy']:
+            raise ConfigError('proxy.type must be set')
+        if tinc['proxy']['type'] == 'socks5': 
+            if 'password' not in tinc['proxy']:
+                raise ConfigError('proxy.password must be set')
+        if tinc['proxy']['type'] == 'socks4' or tinc['proxy']['type'] == 'socks5':            
+            if 'username' not in tinc['proxy']:
+                raise ConfigError('proxy.username must be set')
+        if tinc['proxy']['type'] == 'exec':
+            if 'exec' not in tinc:
+                raise ConfigError('proxy.exec must be set')
+    
+    if 'connect' in tinc:
+        for conn in tinc['connect']:
+            if 'conf_path' in tinc['connect'][conn] and not os.path.exists(tinc['connect'][conn]['conf_path']):
+                raise ConfigError('The configuration file connected to peer {peer} is missing'.format(peer = conn))
+    
+    verify_dhcpv6(tinc)
+    verify_address(tinc)
+    verify_vrf(tinc)
+    
+    if {'is_bond_member', 'mac'} <= set(tinc):
+        print(f'WARNING: changing mac address "{mac}" will be ignored as "{ifname}" '
+              f'is a member of bond "{is_bond_member}"'.format(**tinc))
+    
+    return None
+
+def create_or_delete_network(tinc, op):
+    if tinc is None:
+        return None
+    interface = tinc['ifname']
+    network = tinc['ifname']
+    system_network_dir = f'/etc/tinc/{network}'
+    tinc_network_dir = tinc['network_dir']
+    if tinc:
```

Comment:
No need to check for `if tinc:` when you already have this above:

```
  if tinc is None:
        return None
```

Reply:
Seems to be lost in late conflict resolution, I've manually restored this and kept the mainline in sync